exp/controllers/awsmachinepool_controller.go

```diff
@@ -298,6 +298,21 @@ func (r *AWSMachinePoolReconciler) reconcileNormal(ctx context.Context, machineP
 		return nil
 	}
 
+	lifecycleHookScope, err := scope.NewLifecycleHookScope(scope.LifecycleHookScopeParams{
```

Comment:
`machinePoolScope` should instead provide an interface function like `func (*FooScope) LifecycleHooks() []AWSLifecycleHook` – instead of introducing a new type that covers both EC2 and EKS based clusters in the same "class"